src/modules/storage/LedgerStorage.ts

```diff
@@ -119,4 +164,380 @@ export class LedgerStorage extends Storages
         });
     }
 
+    /**
+     * Put a transaction to database
+     * @param data a transaction data
+     * @param callback If provided, this function will be called when
+     * the database was finished successfully or when an error occurred.
+     * The first argument is an error object.
+     */
+    private putTransaction (block_height: number, tx: any, tx_index: number,
+        tx_hash: string, callback?: (err: Error | null) => void)
+    {
+        if (tx == undefined) {
+            if (callback != undefined)
+                callback(new Error("Parameter validation failed."));
+            return;
+        }
+
+        let sql: string =
+        `INSERT INTO transactions
+            (block_height, tx_index, tx_hash, type, inputs_count, outputs_count)
+        VALUES
+            (?, ?, ?, ?, ?, ?)`;
+        this.db.run(sql,
+            [
+                block_height,
+                tx_index,
+                tx_hash,
+                tx.type,
+                tx.inputs.length,
+                tx.outputs.length
+            ], (err: Error | null) =>
+        {
+            if (err)
+            {
+                if (callback != undefined)
+                    callback(err);
+                return;
+            }
+
+            this.putInputs(block_height, tx, tx_index, tx_hash, (err2: any) =>
+            {
+                if (err2)
```

Comment:
```suggestion
                if (err2 != null)
```

Reply:
if (!err) "if not error..." This expression is more intuitive here. So I prefer this.